changelog/issue-6954.md

```diff
@@ -0,0 +1,4 @@
+level: patch
+reference: issue 6954
+---
+API to fetch github badge which makes two different api call to github service: status and check-run api call; both api's are configured to only retry one time.
```

Comment:
```suggestion
Fixes an issue with github badges that timed out on non-existing branches.
```

This message will end up in the changelog, so it would be good to imagine to be someone who reads the changelog, and needs to understand what was fixed here.